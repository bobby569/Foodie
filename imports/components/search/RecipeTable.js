import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Table, Tag } from 'antd';

const columns = [
	{
		title: 'Dish Name',
		dataIndex: 'label',
		key: 'label',
		render: (text, record) => <Link to={`/recipe/${record.id}`}>{text}</Link>
	},
	{
		title: 'Calories',
		dataIndex: 'calories',
		key: 'calories',
		render: text => ~~text
	},
	{
		title: 'Health',
		dataIndex: 'healthLabels',
		key: 'healthLabels',
		render: text =>
			text.map(item => (
				<Tag key={item} color="green">
					{item}
				</Tag>
			))
	},
	{
		title: 'Diet',
		dataIndex: 'dietLabels',
		key: 'dietLabels',
		render: text =>
			text.map(item => (
				<Tag key={item} color="green">
					{item}
				</Tag>
			))
	}
];

class RecipeTable extends Component {
	render() {
		const { data } = this.props;
		console.log(data);
		return (
			<div className="table">
				<Table
					columns={columns}
					dataSource={data}
					rowKey={record => record.id}
					pagination={false}
					bordered
				/>
			</div>
		);
	}
}

RecipeTable.propTypes = {
	data: PropTypes.array.isRequired
};

export default RecipeTable;
