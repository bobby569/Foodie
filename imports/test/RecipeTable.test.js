import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import RecipeTable from '../components/search/RecipeTable';

describe('RecipeTable', () => {
	const data = [];
	const recipe_table = shallow(<RecipeTable data={data} />);
	const rendered = renderer.create(<RecipeTable data={data} />);

	it('Renders correctly', () => {
		expect(recipe_table.exists()).toBe(true);
		expect(rendered.toJSON()).toMatchSnapshot();
	});
});
