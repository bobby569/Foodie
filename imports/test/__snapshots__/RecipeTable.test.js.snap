// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeTable Renders correctly 1`] = `
<div
  className="table"
>
  <div
    className="ant-table-wrapper"
    style={undefined}
  >
    <div
      className="ant-spin-nested-loading"
      style={null}
    >
      <div
        className="ant-spin-container"
      >
        <div
          className="ant-table ant-table-large ant-table-bordered ant-table-empty ant-table-scroll-position-left"
          id={undefined}
          style={Object {}}
        >
          <div
            className="ant-table-content"
          >
            <div
              className="ant-table-body"
              onScroll={[Function]}
              style={Object {}}
            >
              <table
                className=""
                style={Object {}}
              >
                <colgroup>
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": undefined,
                        "width": undefined,
                      }
                    }
                  />
                </colgroup>
                <thead
                  className="ant-table-thead"
                >
                  <tr
                    style={
                      Object {
                        "height": null,
                      }
                    }
                  >
                    <th
                      className=""
                    >
                      <span>
                        Dish Name
                      </span>
                    </th>
                    <th
                      className=""
                    >
                      <span>
                        Calories
                      </span>
                    </th>
                    <th
                      className=""
                    >
                      <span>
                        Health
                      </span>
                    </th>
                    <th
                      className=""
                    >
                      <span>
                        Diet
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  className="ant-table-tbody"
                />
              </table>
            </div>
            <div
              className="ant-table-placeholder"
            >
              No data
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
